File: src/main/java/com/github/jactor/persistence/JactorPersistence.java
package com.github.jactor.persistence;

import static java.util.Arrays.stream;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.ApplicationContext;

@SpringBootApplication
public class JactorPersistence {

  private static final Logger LOGGER = LoggerFactory.getLogger(JactorPersistence.class);

  static void inspect(ApplicationContext applicationContext, String[] args) {
    if (LOGGER.isDebugEnabled()) {
      boolean noArgs = args == null || args.length == 0;
      String arguments = noArgs ? "without arguments!" : "with arguments: " + String.join(" ", args) + '!';

      LOGGER.debug("Starting application {}", arguments);

      SpringBeanNames springBeanNames = new SpringBeanNames();
      stream(applicationContext.getBeanDefinitionNames()).sorted().forEach(springBeanNames::add);

      LOGGER.debug("Available beans (only simple names):");
      springBeanNames.listBeanNames().stream().map(name -> "- " + name).forEach(LOGGER::debug);

      LOGGER.debug("Ready for service...");
    }
  }

  public static void main(String... args) {
    SpringApplication.run(JactorPersistence.class, args);
  }
}

File: src/main/kotlin/com/github/jactor/persistence/JactorPersistence.kt
package com.github.jactor.persistence

import org.slf4j.LoggerFactory
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.context.ApplicationContext

@SpringBootApplication
class JactorPersistence {

    companion object {
        @JvmStatic
        private val LOGGER = LoggerFactory.getLogger(JactorPersistence::class.java)

        fun inspect(applicationContext: ApplicationContext, args: Array<String>) {
            if (LOGGER.isDebugEnabled) {
                val arguments = if (args.isEmpty()) "without arguments!" else "with arguments: ${args.joinToString { " " }}!"

                LOGGER.debug("Starting application {}", arguments)

                val springBeanNames = SpringBeanNames()
                applicationContext.beanDefinitionNames.sorted().forEach(springBeanNames::add)

                LOGGER.debug("Available beans (only simple names):")
                springBeanNames.listBeanNames().forEach {
                    LOGGER.debug("- $it")
                }

                LOGGER.debug("Ready for service...")
            }
        }

        @JvmStatic
        fun main(args: Array<String>) {
            runApplication<JactorPersistence>(*args)
        }
    }

    private data class SpringBeanNames(
        private val beanNames: MutableList<String> = ArrayList(),
        private val tenNames: MutableList<String> = ArrayList()
    ) {

        fun add(name: String) {
            if (name.contains(".")) {
                val index = name.lastIndexOf('.')
                tenNames.add(name.substring(index + 1))
            } else {
                tenNames.add(name)
            }

            if (tenNames.size == 10) {
                beanNames.add(tenNames.joinToString(", "))
                tenNames.clear()
            }
        }

        fun listBeanNames(): List<String> {
          beanNames.add(tenNames.joinToString(", "))
          tenNames.clear()

          return beanNames
        }
    }
}
